scripts/stylable-metadata-plugin.js

```diff
@@ -0,0 +1,175 @@
+const path = require('path')
+const fs = require('fs')
+const pathFinder = require('react-autodocs-utils/src/path-finder');
+const parse = require('react-autodocs-utils/src/parser/parse');
+const visit = require('react-autodocs-utils/src/parser/visit');
+const types = require('@babel/types');
+const {safeParse, StylableProcessor, processNamespace} = require('@stylable/core')
+
+const parseStylable = source =>
+  new StylableProcessor(undefined, processNamespace).process(safeParse(source)).rawAst
+
+const getImportedStylablePath = ast => new Promise(resolve => {
+  visit(ast)({
+    enter(path) {
+      if (types.isImportDeclaration(path.node)) {
+        const source = path.node.source.value
+
+        if (source.endsWith('.st.css')) {
+          resolve(source)
+          path.stop()
+        }
+      }
+    }
+  })
+
+  resolve(null)
+})
+
+const extractStylableVarStructure = ast =>
+  [].concat.apply(
```

Comment:
perhaps you want a `reduce` here? `.apply` is a rare thing these days, but maybe it's here for a reason?

Reply:
What I would really like to use is not reduce but flat(). However it appeared in node 11 and since it is relatively new, just in case decided not to use it yet.

Reply:
wix is on node 12, also in CI. It should be just fine to use `Array.prototype.flat`